core/src/com/unciv/ui/screens/civilopediascreen/CivilopediaSearchPopup.kt

```diff
@@ -0,0 +1,178 @@
+import com.unciv.models.translations.tr
+import com.unciv.ui.components.ExpanderTab
+import com.unciv.ui.components.UncivTextField
+import com.unciv.ui.components.extensions.disable
+import com.unciv.ui.components.extensions.enable
+import com.unciv.ui.components.extensions.toLabel
+import com.unciv.ui.components.input.KeyCharAndCode
+import com.unciv.ui.components.input.onClick
+import com.unciv.ui.popups.Popup
+import com.unciv.ui.popups.ToastPopup
+import com.unciv.ui.screens.basescreen.BaseScreen
+import com.unciv.ui.screens.basescreen.TutorialController
+import com.unciv.utils.Concurrency
+import com.unciv.utils.launchOnGLThread
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.isActive
+import com.badlogic.gdx.utils.Array as GdxArray
+
+class CivilopediaSearchPopup(
+    private val pediaScreen: CivilopediaScreen,
+    private val tutorialController: TutorialController,
+    private val linkAction: (String) -> Unit
+) : Popup(pediaScreen) {
+    private var ruleset = pediaScreen.ruleset
+    private val searchText = UncivTextField.create("Search text")
+    private val modSelect = ModSelectBox()
+    private lateinit var resultExpander: ExpanderTab
+    private val resultCell: Cell<Actor?>
+    private val searchButton: TextButton
+
+    private var searchJob: Job? = null
+    private var checkLine: (String) -> Boolean = { _ -> false }
+
+    init {
+        searchText.maxLength = 100
+
+        add("{Search text}:".toLabel())
+        add(searchText).growX().row()
+        add("Mod filter:".toLabel())
+        add(modSelect).growX().row()
+        resultCell = add().colspan(2).growX()
+        row()
+
+        searchButton = addButton("Search!", KeyCharAndCode.RETURN) {
+            startSearch(searchText.text)
+        }.actor
+        addCloseButton()
+        showListeners.add {
+            keyboardFocus = searchText
+            searchText.selectAll()
+        }
+        closeListeners.add {
+            if (isSearchRunning()) searchJob!!.cancel()
+        }
+    }
+
+    private fun isSearchRunning() = searchJob?.isActive == true
+
+    private fun startSearch(text: String) {
+        searchButton.disable()
+
+        @Suppress("LiftReturnOrAssignment")
+        if (text.isEmpty()) {
+            checkLine = { true }
+        } else if (".*" in text || '\\' in text || '|' in text) {
+            try {
+                val regex = Regex(text, RegexOption.IGNORE_CASE)
+                checkLine = { regex.containsMatchIn(it) }
+            } catch (ex: Exception) {
+                ToastPopup("That's an invalid regular expression.", pediaScreen, 4000).open(true)
+                searchButton.enable()
+                return
+            }
+        } else {
+            val words = text.split(' ').toSet()
+            checkLine = { line -> words.all { line.contains(it, ignoreCase = true) } }
+        }
+
+        ruleset = modSelect.selectedRuleset() ?: return
```

Comment:
You re-enable is on line 82, but not here - intentional since there is no ruleset?